### PR TITLE
Refactor to support passphrases.

### PR DESCRIPTION
**Pull request by jaypha**

### CODE REVIEW

File: classes/local/step/gpg_trait.php [hunk 1]
```diff
@@ -114,29 +110,50 @@ public function execute($input = null) {
...
+            $options[] = '--homedir ' . $homedir;
+        }
+
+        // The passphrase needs to be piped into the stdin of the gpg command.
+        $pipedpassphrase = '';
+        // Because variables use redacted values by default, we evaluate the passphrase explicitly.
+        $passphrase = $this->get_variables()->evaluate($this->stepdef->config->passphrase);
+        if ($passphrase) {
+            $options[] = '--pinentry-mode loopback --passphrase-fd 0';
+            $pipedpassphrase = 'echo "' . $passphrase . '" | ';
```
Comment (brendanheywood) [inline classes/local/step/gpg_trait.php:131]: should this use escapeshellarg?

Comment (brendanheywood) [reply classes/local/step/gpg_trait.php:131]: I suspect escapeshellarg is missing in a few other existing places too

Comment (jaypha) [reply classes/local/step/gpg_trait.php:131]: Yes, it should.

Comment (jaypha) [reply classes/local/step/gpg_trait.php]: done

**Review by keevan — APPROVED** [review]
Looks good to me, can you add a version bump please

**Pull request merged**